commit 31cb8a8f44664a7771e8eba64140d165999d8af3
Author: Lukas Tinkl <lukas@kde.org>
Date:   Tue Nov 15 18:43:29 2011 +0100

    unify source names, it might confuse the battery plasmoid when devices get added/removed
    
    (possible fix for https://bugzilla.redhat.com/show_bug.cgi?id=753429)

diff --git a/plasma/generic/dataengines/powermanagement/powermanagementengine.cpp b/plasma/generic/dataengines/powermanagement/powermanagementengine.cpp
index c80c917..a5344da 100644
--- a/plasma/generic/dataengines/powermanagement/powermanagementengine.cpp
+++ b/plasma/generic/dataengines/powermanagement/powermanagementengine.cpp
@@ -141,7 +141,7 @@ bool PowermanagementEngine::sourceRequestEvent(const QString &name)
             }
         }
 
-        setData("Battery", "Has battery", !batterySources.isEmpty());
+        setData("Battery", "Has Battery", !batterySources.isEmpty());
         if (!batterySources.isEmpty()) {
             setData("Battery", "Sources", batterySources);
             QDBusMessage msg = QDBusMessage::createMethodCall("org.kde.Solid.PowerManagement",
@@ -238,7 +238,7 @@ void PowermanagementEngine::deviceRemoved(const QString& udi)
 
         QStringList sourceNames(m_batterySources.values());
         sourceNames.removeAll(source);
-        setData("Battery", "sources", sourceNames);
+        setData("Battery", "Sources", sourceNames);
     }
 }
 
@@ -272,7 +272,7 @@ void PowermanagementEngine::deviceAdded(const QString& udi)
             updateBatteryChargePercent(battery->chargePercent(), device.udi());
             updateBatteryPlugState(battery->isPlugged(), device.udi());
 
-            setData("Battery", "sources", sourceNames);
+            setData("Battery", "Sources", sourceNames);
         }
     }
 }

commit 44949c24779ee330917684d77fc7dc989fe4eaa6
Author: Lukas Tinkl <lukas@kde.org>
Date:   Thu Nov 17 12:41:39 2011 +0100

    unbreak signalling remaining time on the battery
    
    one shouldn't trust the upower XML introspection :/

diff --git a/powerdevil/daemon/backends/upower/dbus/org.freedesktop.UPower.xml b/powerdevil/daemon/backends/upower/dbus/org.freedesktop.UPower.xml
index a3f0004..807f4e6 100644
--- a/powerdevil/daemon/backends/upower/dbus/org.freedesktop.UPower.xml
+++ b/powerdevil/daemon/backends/upower/dbus/org.freedesktop.UPower.xml
@@ -54,7 +54,7 @@ method return sender=:1.386 -> dest=:1.451 reply_serial=2
     <!-- ************************************************************ -->
 
     <signal name="DeviceAdded">
-      <arg name="device" type="o">
+      <arg name="device" type="s">
         <doc:doc><doc:summary>Object path of device that was added.</doc:summary></doc:doc>
       </arg>
 
@@ -70,7 +70,7 @@ method return sender=:1.386 -> dest=:1.451 reply_serial=2
     <!-- ************************************************************ -->
 
     <signal name="DeviceRemoved">
-      <arg name="device" type="o">
+      <arg name="device" type="s">
         <doc:doc><doc:summary>Object path of device that was removed.</doc:summary></doc:doc>
       </arg>
 
@@ -86,7 +86,7 @@ method return sender=:1.386 -> dest=:1.451 reply_serial=2
     <!-- ************************************************************ -->
 
     <signal name="DeviceChanged">
-      <arg name="device" type="o">
+      <arg name="device" type="s">
         <doc:doc><doc:summary>Object path of device that was changed.</doc:summary></doc:doc>
       </arg>
 
